service/gcs/core/gcs/storage.go

```diff
@@ -331,7 +330,7 @@ func (c *gcsCore) writeConfigFile(id string, config oci.Spec) error {
 	}
 	defer configFile.Close()
 	writer := bufio.NewWriter(configFile)
-	if err := json.NewEncoder(writer).Encode(config); err != nil {
+	if err := utils.EncodeJSONWithHresult(writer, config); err != nil {
```

Comment:
I'm not sure if we want to use these methods _everywhere_ we deal with JSON. The HRESULT probably should mean something was wrong with the JSON that was passed to GCS from HCS. But in this case, for example, the problem is presumably a logic bug that we can't actually encode an oci.Spec object.